### PR TITLE
docs: verifier crate docs

### PR DESCRIPTION
**Pull request by yuwen01**
Documents the verifier crate. Assumes that v3.2.0 is released and that docs.rs contains information on `sp1-verifier`.

### CODE REVIEW

File: book/verification/off-chain-verification.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+you must input the `sp1_proof_with_public_values.bytes()`, `sp1_proof_with_public_values.public_values.to_vec()`,
+and the vkey hash manually. For an example of how to do this, see [Wasm Verification](#wasm-verification).
+
+## Wasm Verification
+
+The [`example-sp1-wasm-verifier`](https://github.com/succinctlabs/example-sp1-wasm-verifier) demonstrates how to
+verify SP1 proofs in wasm. For a more detailed explanation of the process, please see the [README](https://github.com/succinctlabs/example-sp1-wasm-verifier/blob/main/README.md).
+
+At a high level, the process is as follows:
+
```
Comment (yuwen01) [inline book/verification/off-chain-verification.md:60]: I want to give some sense of how to do wasm verification, but the process is kind of complicated. 

See [here](https://github.com/succinctlabs/example-sp1-wasm-verifier/pull/3/files#diff-13ee4b2252c9e516a0547f2891aa2105c3ca71c6d7a1e682c69be97998dfc87e) for the entire process.

File: book/verification/off-chain-verification.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+## Rust `no_std` Verification
+
+You can verify SP1 Groth16 and Plonk proofs in `no_std` environments with [`sp1-verifier`](https://docs.rs/sp1-verifier/latest/sp1_verifier/).
+
+`sp1-verifier` is also patched to verify Groth16 and Plonk proofs within the SP1 ZKVM, using
+[bn254](https://blog.succinct.xyz/succinctshipsprecompiles/) precompiles. For an example of this, see
+the [Groth16 Example](https://github.com/succinctlabs/sp1/tree/main/examples/groth16/).
+
+### Instafllation
```
Comment (jtguibas) [inline book/verification/off-chain-verification.md:11]: typo

File: book/verification/off-chain-verification.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+let sp1_proof_with_public_values = SP1ProofWithPublicValues::load(proof_file).unwrap();
+
+let proof = sp1_proof_with_public_values.bytes();
+let public_inputs = sp1_proof_with_public_values.public_values.to_vec();
+
+// This vkey hash was derived by calling `vk.bytes32()` on the verifying key.
+let vkey_hash = "0x00e60860c07bfc6e4c480286c0ddbb879674eb47f84b4ef041cf858b17aa0ed1";
+
+let is_valid =
+    crate::Groth16Verifier::verify(&proof, &public_inputs, vkey_hash, &crate::GROTH16_VK_BYTES)
```
Comment (jtguibas) [inline book/verification/off-chain-verification.md:46]: you should show a more detailed example that includes also the import paths for these variables

Comment (jtguibas) [reply book/verification/off-chain-verification.md:46]: the crate:: is unrealistic

File: book/verification/off-chain-verification.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+Note that the SP1 SDK itself is *not* `no_std` compatible. To use the verifier crate in fully `no_std` environments,
+you must input the `sp1_proof_with_public_values.bytes()`, `sp1_proof_with_public_values.public_values.to_vec()`,
+and the vkey hash manually. For an example of how to do this, see [Wasm Verification](#wasm-verification).
+
+## Wasm Verification
+
+The [`example-sp1-wasm-verifier`](https://github.com/succinctlabs/example-sp1-wasm-verifier) demonstrates how to
+verify SP1 proofs in wasm. For a more detailed explanation of the process, please see the [README](https://github.com/succinctlabs/example-sp1-wasm-verifier/blob/main/README.md).
+
+At a high level, the process is as follows:
```
Comment (jtguibas) [inline book/verification/off-chain-verification.md:59]: if you're gonna defer the docs to that repo, just delete this part imo.

**Review by jtguibas — APPROVED** [review]

**Pull request merged**